src/main/java/org/craftercms/deployer/impl/processors/AbstractDeploymentProcessor.java

```diff
@@ -129,6 +130,15 @@ public boolean supportsMode(Deployment.Mode mode) {
 
     @Override
     public void execute(Deployment deployment) {
+        try {
+            this.currentDeployment = deployment;
```

Comment:
Not sure this is a good idea right now. I'm almost sure this is ok since processor instances are target specific and a target just runs a processor at a time, but if we're wrong we can break a lot of things